integreat_cms/cms/views/contacts/contact_bulk_actions.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from django.contrib import messages
+from django.http import Http404
+from django.utils.translation import gettext_lazy as _
+from django.utils.translation import ngettext_lazy
+from django.views.generic.list import MultipleObjectMixin
+
+from ...models import Contact
+from ...utils.stringify_list import iter_to_string
+from ...utils.translation_utils import gettext_many_lazy as __
+from ..bulk_action_views import BulkActionView
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from django.http import HttpRequest, HttpResponse
+    from django.http.response import HttpResponseRedirect
+
+logger = logging.getLogger(__name__)
+
+
+class ContactBulkAction(BulkActionView):
+    """
+    Class to overwrite get_queryset()
+    """
+
+    def get_queryset(self) -> Any:
+        r"""
+        This method overrides get_queryset from BulkActionView to handle the relation between contact and poi
+
+        :raises ~django.http.Http404: HTTP status 404 if no objects with the given ids exist
+
+        :return: The QuerySet of the filtered links
+        """
+        # This workaround is necessary to enable the async tests for the SUMM.AI client
+        logger.debug("request body: %s", self.request.body)
+        queryset = (
+            super(BulkActionView, self)
+            .get_queryset()
+            .filter(
+                poi__region=self.request.region,
+                id__in=self.request.POST.getlist("selected_ids[]"),
+            )
+        )
+        if not queryset:
+            raise Http404(f"No {self.model._meta.object_name} matches the given query.")
+        return queryset
```

Comment:
IMO a super call with parameters is uncommon and can be a bit confusing. What do you think about adding a new function like `def get_extra_filters(self) -> Q` to `BulkActionView`?
```python
class BulkActionView(...):
    ...

    def get_queryset(...):
        ...
        queryset = (
            super()
                .get_queryset()
                .filter(
                    id__in=self.request.POST.getlist("selected_ids[]"),
                )
                .filter(self.get_extra_filters())
        )
        ...

    def get_extra_filters(self) -> Q:
       return Q(region=self.request.region)
```

Then you would just have to overwrite this method in `ContactBulkAction` and would also reduce code duplication a bit

Reply:
Thanks for the suggestion. I think I will try it as you suggest. I agree that it's probably less confusing.